hpcgap/lib/oper.g

```diff
@@ -1179,37 +1183,34 @@ BIND_GLOBAL( "OPER_SetupAttribute", function(getter, flags, mutflag, filter, ran
           return;
       end);
 
+# construct getter, setter and tester
+BIND_GLOBAL( "NewAttribute", function ( name, filter, args... )
+    local  flags, mutflag, getter, rank;
 
-BIND_GLOBAL( "NewAttribute", function ( arg )
-    local   name, filter, flags, mutflag, getter, rank;
-
-    # construct getter, setter and tester
-    name   := arg[1];
-    filter := arg[2];
+    mutflag := LEN_LIST(args) >= 1 and (args[1] = "mutable" or args[1] = true);
+    if LEN_LIST(args) = 1 and IS_INT(args[1]) then
+        rank := args[1];
+    elif LEN_LIST(args) = 2 and IS_INT(args[2]) then
+        rank := args[2];
+    else
+        rank := 1;
+    fi;
 
     if not IS_OPERATION( filter ) then
       Error( "<filter> must be an operation" );
     fi;
     flags:= FLAGS_FILTER( filter );
 
-    # the mutability flags is the third one (which can also be the rank)
-    mutflag := LEN_LIST(arg) = 3 and arg[3] = "mutable";
-
     # construct a new attribute
     if mutflag then
         getter := NEW_MUTABLE_ATTRIBUTE( name );
     else
         getter := NEW_ATTRIBUTE( name );
     fi;
-    if LEN_LIST(arg) = 3 and IS_INT(arg[3]) then
-        rank := arg[3];
-    else
-        rank := 1;
-    fi;
     STORE_OPER_FLAGS(getter, [ flags ]);
 
     atomic FILTER_REGION do
-    OPER_SetupAttribute(getter, flags, mutflag, filter, rank, name);
+        OPER_SetupAttribute(getter, flags, mutflag, filter, rank, name);
```

Comment:
I deliberately removed indents like that to reduce spurious diffs without whitespace suppression. Granted, in this particular case it matters little, but for consistency, I'd like to keep it as it was.